doc/api.md

```diff
@@ -2,31 +2,89 @@
 
 ***If you are running the server and want to enable the API, set the environment variable `API_ENABLED=true`.***
 
-This API is currently in a very limited alpha, and is not enabled in any of our deployments. Note that this API may change at any time, without warning.
+The API design is evolving. Backwards-incompatible changes will result in a version change.
+
+The lastest version is *v2*.
+
+## Clients
+
+See e.g. https://github.com/18F/c2-api-client-ruby for a ready-to-use API client. It handles
+all the authentication for you.
+
+## Authentication
+
+Authentication is provided with OAuth2. You must:
+
+* create an application with OAuth keys via https://cap.18f.gov/oauth/applications
+* authorize your new application (click the **Authorize** button)
+* use your OAuth keys to obtain a Bearer auth token
+* include your Bearer auth token with every API request
+
+Example:
+
+```bash
+% export MY_OAUTH_KEY=your-key-here
+% export MY_OAUTH_SECRET=your-secret-here
+% export MY_CREDS=`echo "$MY_OAUTH_KEY:$MY_OAUTH_SECRET" | base64`
+% curl -i -X POST -H "Authorization: Basic $MY_CREDS" \
+  -d 'grant_type=client_credentials' \
+  https://cap.18f.gov/oauth/token
+```
+
+Take note of the `access_token` string value in the response. You will use it below.
 
 ## Schemas
 
 * All decimals are strings ([more info](https://github.com/rails-api/active_model_serializers/issues/202))
 * All times are in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) format
 
-### Step
+### Proposal
+
+Each proposal object is composed of a few attributes and some child objects.
+
+### Attributes
 
 Attribute | Type | Note
 --- | --- | ---
-`id` | integer |
+`id` | integer | primary identifier
+`status` | string | one of: `pending`, `cancelled`, `complete`
+`created_at` | string | timestamp
+`updated_at` | string | timestamp
+`client_data_type` | string | name of the client data class
+
+### Requester
+
+Who created the proposal. Refers to a [User](#user) record.
+
+Attribute | Type | Note
+--- | --- | ---
+`id` | integer | primary identifier
+`created_at` | string | timestamp
+`updated_at` | string | timestamp
+
+### Steps
+
+Array of step objects.
+
+Attribute | Type | Note
+--- | --- | ---
+`id` | integer | primary identifier
 `status` | string | Can be `pending`, `actionable`, `approved`, or `canceled`
-`user` | [User](#user) | a.k.a. "the approver"
+`user` | [User](#user) | who needs to complete the step
```

Comment:
looks like we deleted the `User` docs below - should w re-add so this is more helpful?